src/deps/zcl_abaplint_deps_find.clas.abap

```diff
@@ -263,59 +398,141 @@ CLASS ZCL_ABAPLINT_DEPS_FIND IMPLEMENTATION.
+        CONTINUE.
+      ENDIF.
       lv_index = sy-tabix.
-      SELECT SINGLE devclass FROM tadir INTO lv_devclass
-        WHERE pgmid = 'R3TR'
-        AND object = ls_tadir-ref_obj_type
-        AND obj_name = ls_tadir-ref_obj_name.
-* todo, should check if its a sub-package
-      IF sy-subrc <> 0 OR lv_devclass CP |{ iv_package }*|.
+      lv_devclass = determine_package( iv_object_type   = ls_tadir-ref_obj_type
+                                       iv_object_name = ls_tadir-ref_obj_name ).
+      lv_flag = 'X'.
+      IF sy-subrc = 0.
+        READ TABLE mv_packages FROM lv_devclass TRANSPORTING NO FIELDS.
+        IF sy-subrc <> 0.
+          CLEAR lv_flag.
+        ENDIF.
+      ENDIF.
+      IF lv_flag = 'X'.
         DELETE lt_tadir INDEX lv_index.
-        CONTINUE.
       ENDIF.
     ENDLOOP.
 
-    APPEND LINES OF lt_tadir TO rt_total.
+    "Add to dependency collection
+    APPEND LINES OF lt_tadir TO mv_results.
 
-    DATA ls_object TYPE zif_abapgit_definitions=>ty_tadir.
-    DATA lv_level LIKE iv_level.
+    SORT mv_results BY ref_obj_type ASCENDING ref_obj_name ASCENDING.
+    DELETE ADJACENT DUPLICATES FROM mv_results COMPARING ref_obj_type ref_obj_name.
 
+*
+* if sap object, do not go deeper
+*
+    IF ( ls_tadir_obj-author = 'SAP'
+        OR ls_tadir_obj-author = 'SAP*' )
+        AND ls_tadir_obj-srcsystem = 'SAP'.
+      RETURN.
+    ENDIF.
+*
+*   Certain types do not have dependent object or are resolved by this call
+*
+    DELETE lt_tadir WHERE ref_obj_type = 'MSAG'. "Message AG
+    DELETE lt_tadir WHERE ref_obj_type = 'DTEL'. "Data Element
+*
+* Try to find dependend objects
+*
     LOOP AT lt_tadir INTO ls_tadir.
+
       ls_object-object   = ls_tadir-ref_obj_type.
       ls_object-obj_name = ls_tadir-ref_obj_name.
 
       lv_level = iv_level + 1.
 
       get_dependencies(
-        iv_package = iv_package
         is_object  = ls_object
         iv_level   = lv_level ).
     ENDLOOP.
 
   ENDMETHOD.
 
 
+  METHOD prepare_supported_types.
+    rs_types-prog = 'X'.
+    rs_types-fugr = 'X'.
+    rs_types-ldba = ' '.
+    rs_types-msag = 'X'.
+    rs_types-tran = 'X'.
+    rs_types-func = 'X'.
+    rs_types-dial = ' '.
+    rs_types-tabl = 'X'.
+    rs_types-shlp = ' '.
+    rs_types-doma = ' '. "Level not required
+    rs_types-dtel = 'X'.
+    rs_types-view = 'X'.
+    rs_types-mcob = ' '.
+    rs_types-mcid = ' '.
+    rs_types-para = 'X'.
+    rs_types-conv = ' '.
+    rs_types-suso = ' '.
+    rs_types-type = 'X'.
+    rs_types-ttyp = 'X'.
+    rs_types-stru = 'X'.
+    rs_types-enqu = ' '.
+    rs_types-sqlt = ' '.
+    rs_types-clas = 'X'.
+    rs_types-intf = 'X'.
+    rs_types-udmo = ' '.
+    rs_types-ueno = ' '.
+    rs_types-shi3 = ' '.
+    rs_types-cntx = ' '.
+    rs_types-ttab = 'X'.
+    rs_types-iasp = ' '.
+    rs_types-iatu = ' '.
+    rs_types-clif = ' '.
+    rs_types-sobj = ' '.
+    rs_types-wdyn = ' '.
+    rs_types-wdya = ' '.
+    rs_types-xslt = ' '.
```

Comment:
we need xslt (also "id" transformation https://github.com/abaplint/abaplint/pull/1025)

Reply:
This has been changed already using this way for each type:
![image](https://user-images.githubusercontent.com/53995991/86774356-c26f0e80-c056-11ea-8608-5c38045459f0.png)
the call now is done once in the constructor.